src/catalog/distribute_iterator.cc

```diff
@@ -284,11 +293,57 @@ void DistributeWindowIterator::Next() {
                 it_->SeekToFirst();
```

Comment:
`it_` should be only updated on success, and do nothing changed on fail.
In case if all `tablets_` left is empty, then try to seek from remote, the `it_` do not to be `nullptr`

Reply:
will check `it_` is `nullptr` and validation in getkey/getvalue function